libopae/src/wsid_list.c

```diff
@@ -116,17 +116,21 @@ bool wsid_del(struct wsid_map **root, uint64_t wsid)
  *
  * @param root
  */
-void wsid_cleanup(struct wsid_map **root)
+void wsid_cleanup(struct wsid_map **root, void (*clean)(struct wsid_map * ))
 {
 	if (!*root)
 		return;
 
 	while ((*root)->next) {
 		struct wsid_map *tmp = *root;
 		*root = (*root)->next;
+		if (clean)
```

Comment:
Aren't you asking for trouble using multi-line without brackets?

Reply:
I guess I'm not seeing it.  Isn't clean(tmp); a single statement?